apps/system/js/updatable.js

```diff
@@ -144,15 +144,53 @@ function SystemUpdatable() {
   this.downloading = false;
   this.paused = false;
   this.showingApplyPrompt = false;
+  this._updateProviderPromise = null;
+  this._idleObserver = null;
 
+  Service.register('updateCheck', this);
+
+  this._initUpdateProvider();
   // XXX: this state should be kept on the platform side
   // https://bugzilla.mozilla.org/show_bug.cgi?id=827090
-  this.checkKnownUpdate(UpdateManager.checkForUpdates.bind(UpdateManager));
-
-  window.addEventListener('mozChromeEvent', this);
+  this.checkKnownUpdate().then((value) => {
+    if (value) {
+      this.updateCheck();
+    }
+  });
 }
 
 SystemUpdatable.KNOWN_UPDATE_FLAG = 'known-sysupdate';
+SystemUpdatable.APPLY_IDLE_TIMEOUT = 600;
+
+SystemUpdatable.prototype._initUpdateProvider = function() {
+  this._getUpdateProvider().then((provider) => {
+    if (provider) {
+      provider.addEventListener('updateavailable', this);
+      provider.addEventListener('updateready', this);
+      provider.addEventListener('progress', this);
+      provider.addEventListener('error', this);
+    }
+  });
+};
+
+SystemUpdatable.prototype._getUpdateProvider = function() {
+  if (!this._updateProviderPromise) {
+    this._updateProviderPromise =
+    window.navigator.updateManager.getProviders().then((providerInfos) => {
```

Comment:
should follow system_update_manage to 
1. getActiveProvider first, then
2. if not exist, choose first provider instead